### PR TITLE
[BCI-1283] Head Broadcaster Generalisation

### PR DESCRIPTION
**Pull request by ExtremelySunnyYK**
This PR aims to generalise the Head Broadcaster component which is responsible for broadcasting head events to internal components and notifies other subscribers.

#### Changes:
- Moved `HeadTrackable` to `common/types` because its used by more than just the `txmgr`
- Created a generalised Head Broadcaster Component.
- Generalised Head Broadcaster component methods.

#### References:
[Ticket](https://smartcontract-it.atlassian.net/browse/BCI-1281?jql=labels%20%3D%20%22intern-yongkang-2023%22)
[Design Proposal](https://github.com/smartcontractkit/CLIPs/pull/68/files?short_path=eb61d5e#diff-eb61d5e7507073832f3bad6081e745f0f4a5dc565ae7c9fe9d04075ac1905ffc)

### CODE REVIEW

**Review by jmank88 — DISMISSED** [review]

File: common/types/mocks/head_trackable.go
Comment (aalu1418) [inline common/types/mocks/head_trackable.go:1]: can you make sure to delete the old `head_trackable` mock too? it is probably under `common/txmgr/types/mocks`

**Review by aalu1418 — COMMENTED** [review]
overall, looks good - just one nit

File: common/types/mocks/head_trackable.go
Comment (ExtremelySunnyYK) [reply common/types/mocks/head_trackable.go:1]: thank you for pointing that out!

Comment (ExtremelySunnyYK) [reply common/types/mocks/head_trackable.go:1]: updated the PR to remove the mock

**Review by aalu1418 — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]

**Review by prashantkumar1982 — APPROVED** [review]

**Pull request merged**